src/m365/pa/commands/app/app-export.ts

```diff
@@ -0,0 +1,238 @@
+      };
+
+      const file = await request.get<string>(requestOptions);
+
+      const path = `${args.options.path ? args.options.path : './'}${args.options.packageDisplayName}.zip`;
+
+      fs.writeFileSync(path, file, 'binary');
+      if (!args.options.path || this.verbose) {
+        if (this.verbose) {
+          logger.logToStderr(`File saved to path '${path}'`);
+        }
+        else {
+          logger.log(path);
+        }
+      }
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getPackageResources(args: CommandArgs, logger: Logger): Promise<any> {
+    if (this.verbose) {
+      logger.logToStderr('Getting the Microsoft Power App resources...');
+    }
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/providers/Microsoft.BusinessAppPlatform/environments/${formatting.encodeQueryParameter(args.options.environment)}/listPackageResources?api-version=2016-11-01`,
+      headers: {
+        accept: 'application/json'
+      },
+      data: {
+        baseResourceIds: [
+          `/providers/Microsoft.PowerApps/apps/${args.options.id}`
+        ]
+      },
+      responseType: 'json'
+    };
+
+    const response = await request.post<any>(requestOptions);
+    Object.keys(response.resources).forEach((key) => {
+      response.resources[key].suggestedCreationType = 'Update';
+    });
+    return response.resources;
+  }
+
+  private async exportPackage(args: CommandArgs, logger: Logger): Promise<string> {
+    if (this.verbose) {
+      logger.logToStderr(`Initiating package export for Microsoft Power App ${args.options.id}...`);
+    }
+
+    const resources = await this.getPackageResources(args, logger);
+
+    const requestOptions: CliRequestOptions = {
+      url: `${this.resource}/providers/Microsoft.BusinessAppPlatform/environments/${formatting.encodeQueryParameter(args.options.environment)}/exportPackage?api-version=2016-11-01`,
+      headers: {
+        accept: 'application/json'
+      },
+      responseType: 'json',
+      data: {
+        includedResourceIds: [
+          `/providers/Microsoft.PowerApps/apps/${args.options.id}`
+        ],
+        details: {
+          creator: args.options.packageCreatedBy,
+          description: args.options.packageDescription,
+          displayName: args.options.packageDisplayName,
+          sourceEnvironment: args.options.packageSourceEnvironment
+        },
+        resources: resources
+      },
+      fullResponse: true
+    };
+
+    const response: any = await request.post<any>(requestOptions);
+
+    const location = response.headers.location;
+
+    if (this.verbose) {
+      logger.logToStderr(`Location of the Power Apps export: ${location}`);
+    }
+
+    return response.headers.location;
+  }
+
+  private async getPackageLink(args: CommandArgs, logger: Logger): Promise<string> {
+    let status;
+    let link;
+
+    const location = await this.exportPackage(args, logger);
```

Comment:
Let's move this function call outside of the `getPackageLink` function. A function should only have one goal and now `getPackageLink` has two.

Let's move the function call to the commandAction method right above the call to `getPackageLink`.